packages/maplestory-open-api-js/src/types/ranking.ts

```diff
@@ -0,0 +1,94 @@
+export interface CommonRankingProps {
+  /** 랭킹 업데이트 일자 (KST, 일 단위 데이터로 시, 분은 일괄 0으로 표기)
+   * example: 2023-12-21T00:00+09:00
+   * */
+  date: string;
+  /** 종합 랭킹 순위 */
+  ranking: number;
+  /** 캐릭터 명 */
+  character_name: string;
+  /** 월드 명 */
+  world_name: string;
+  /** 직업 명 */
+  class_name: string;
+  /** 전직 직업 명 */
+  sub_class_name: string;
+}
```

Comment:
```suggestion
export interface CommonRankingProps<T extends Record<string, unkown>> {
  /** 랭킹 업데이트 일자 (KST, 일 단위 데이터로 시, 분은 일괄 0으로 표기)
   * example: 2023-12-21T00:00+09:00
   * */
  date: string;
  /** 종합 랭킹 순위 */
  ranking: number;
  /** 캐릭터 명 */
  character_name: string;
  /** 월드 명 */
  world_name: string;
  /** 직업 명 */
  class_name: string;
  /** 전직 직업 명 */
  sub_class_name: string;
} & T
```

느낌으로도 작성할 수 있을 것 같네요~!